### ISSUE TITLE
Lower compiler version requirement for xxh3

### ISSUE THREAD

**Issue by domenukk** [DoumanAsh/xxhash-rust#1]
Looking for a working xxh3 implementation that does not have to build c code, I stumbled upon this repo.
Sadly, the build aborts with 
```
error[E0658]: `if` is not allowed in a `const fn`
  --> .cargo/registry/src/github.com-1ecc6299db9ec823/xxhash-rust-0.8.0-beta.3/src/xxh3_utils.rs:10:5
   |
10 | /     if seed == 0 {
11 | |         return DEFAULT_SECRET;
12 | |     }
   | |_____^
   |
   = note: see issue #49146 <https://github.com/rust-lang/rust/issues/49146> for more information

error[E0658]: `while` is not allowed in a `const fn`
  --> .cargo/registry/src/github.com-1ecc6299db9ec823/xxhash-rust-0.8.0-beta.3/src/xxh3_utils.rs:30:5
   |
30 | /     while idx < NB_ROUNDS {
31 | |         let lo = read_u64(&DEFAULT_SECRET, idx * 16).wrapping_add(seed).to_le_bytes();
32 | |         let hi = read_u64(&DEFAULT_SECRET, idx * 16 + 8).wrapping_sub(seed).to_le_bytes();
33 | |
...  |
52 | |         idx += 1;
53 | |     }
   | |_____^
   |
   = note: see issue #52000 <https://github.com/rust-lang/rust/issues/52000> for more information

error: aborting due to 2 previous errors
```

It is my understanding, that the const_ version is the one that would need this feature from nightly, but maybe it can be dropped for the non-const versions of this lib?

**Comment by DoumanAsh** [comment]
It is true that this could be removed as this utility while useful, is not really necessary for non-const version.
I added primarily so that user could generate secret at compile time instead of using `_seed` algorithm.
But we can remove it from non-const version too

**Comment by DoumanAsh** [comment]
Done https://github.com/DoumanAsh/xxhash-rust/commit/1d4153cb779bee353c5a977ba2b9bf85389e9abc
Please test that crate works on your stable

**Comment by DoumanAsh** [comment]
I published `v0.8.0-beta.4`
Use it

**Comment by domenukk** [comment]
Thank you! :)
